script/single-deployment/CowAmmModule.s.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.8.0 <0.9.0;
+
+import {console} from "forge-std/Script.sol";
+
+import {CowAmmModule} from "src/CowAmmModule.sol";
+
+import {Utils} from "script/libraries/Utils.sol";
+
+contract DeployCowAmmModule is Utils {
+    address internal handler;
+
+    constructor() {
+        solutionSettler = addressEnvOrDefault("SETTLEMENT_CONTRACT", DEFAULT_SETTLEMENT_CONTRACT);
+        extensibleFallbackHandler =
+            addressEnvOrDefault("EXTENSIBLE_FALLBACK_HANDLER_CONTRACT", DEFAULT_EXTENSIBLE_FALLBACK_HANDLER_CONTRACT);
+        composableCow = addressEnvOrDefault("COMPOSABLE_COW_CONTRACT", DEFAULT_COMPOSABLE_COW_CONTRACT);
+        handler = addressEnvOrDefault("CONSTANT_PRODUCT_CONTRACT", address(0));
+
+        console.log("Settlement contract at %s.", solutionSettler);
+        console.log("ExtensibleFallbackHandler contract at %s.", extensibleFallbackHandler);
+        console.log("ComposableCoW contract at %s.", composableCow);
+
+        assertHasCode(solutionSettler, "no code at expected solution settler contract");
+        assertHasCode(extensibleFallbackHandler, "no code at expected extensible fallback handler contract");
+        assertHasCode(composableCow, "no code at expected composable cow contract");
+
+        // Special case: if the constant product contract is not provided, we assume
+        // that this script is being called from DeployAllContracts and that the
+        // constant product contract will be deployed by it.
+        if (handler == address(0)) {
+            console.log("No constant product contract provided, assuming called in deployAll.");
+        } else {
+            console.log("handler contract at %s.", handler);
+            assertHasCode(handler, "no code at expected handler contract");
+        }
+    }
+
+    function run() public virtual {
+        if (handler == address(0)) {
+            console.log("No constant product contract provided in single deployment!");
+            console.log(
+                "Please provide the constant product contract address in the environment variable CONSTANT_PRODUCT_CONTRACT."
+            );
+            revert();
+        }
```

Comment:
A nice feature for users: you can read the contract address from `networks.json`. I did something similar [in my deploy script attempts](https://github.com/cowprotocol/cow-amm/blob/4db6a8cd660ad594e544512749931523d9328498/script/SetupOrder.s.sol#L118-L122), it should be easier to adapt this function to be used here. In case, note the changes to `foundry.toml`!

The drawback is that users could deploy manually `ConstantProduct` and then `CowAmmModule` without updating networks.json in between.

Reply:
Yes, I think that given the possibility of manual deployment, it may make sense to keep this as an explicit requirement for them to set.